File: platform/Leetcode/src/RemoveAdjacentDuplicates.java
import java.util.*;

public class RemoveAdjacentDuplicates{
    
    public static void main(String[] args){
        System.out.print("Enter a string of alphabets and I'll remove the adjacent duplicates: ");
        Scanner sc = new Scanner(System.in);
        String input = sc.nextLine();
        String output = removeDuplicates(input);              
        System.out.println("The string without adjacent duplicates is: " + output); 
        
    }
    
    public static String removeDuplicates(String input){

        char[] charArray = input.toCharArray();
        Stack<Character> stack = new Stack<Character>();

        for (char c :charArray) {
            if (!stack.isEmpty() && stack.peek() == c) {
                stack.pop();
            } else {
                stack.push(c);
            }
        }

        StringBuilder sb = new StringBuilder();
        for (char each: stack){
            sb.append(each);
        }

        return sb.toString();
    }
}

File: platform/Leetcode/src/remove_adjacent_duplicates.py
def remove_duplicates_stackwise(string):
    assert isinstance(string, str), "Input should be a string"
    stack = []
    for char in string:
        if stack and stack[-1] == char:
            stack.pop()
        else:
            stack.append(char)

    return ''.join(stack)

def main():
    user_input = input("Enter a string of alphabets with duplicates (e.g., aabbacd): ")
    output = remove_duplicates_stackwise(user_input)
    print("The remaining string after removing duplicates is: ", output)
    
if __name__ == '__main__':
    main()
